feat: add ARM64 support

bweigel opened the pull request.

bweigel committed:
use SAM build image

mnai-prout-james commented:
@bweigel - thanks for this, even though still in draft.

I got this to work locally via clang instead of gss compiler, which appears to be the recommendation from the tesseract team too.

Edits were:
> RUN yum -y install clang gcc-c++ make autoconf aclocal automake libtool \

and:

> RUN echo "/opt/leptonica/lib" > /etc/ld.so.conf.d/leptonica.conf
> RUN echo sudo ldconfig

I also stripped out CXXFLAGS on the tesseract build

It took an age to build, but that could be my laptop. Either way - I've just published to my AWS account as a layer and working just fine, thanks!

Commit:
add some docs & refactor dockerimage

Commit:
trial for some ci

Commit:
add compiled binaries (CLANG)

Commit:
update example code to cdk2 & add arm funciton

Commit:
fix

Commit:
update example deployment

Commit:
unfortunately only works with the manylinuy argument ...

https://aws.amazon.com/premiumsupport/knowledge-center/lambda-python-package-compatible/
